### ISSUE TITLE
import no longer supported with Terraform 0.12.x as of v1.29.0

### ISSUE THREAD

**Issue by blbradley** [terraform-aws-modules/terraform-aws-lambda#83]
on Terraform 0.12.16 and 0.12.29 with v1.31.0

```
✗ terraform import "module.lambda.aws_lambda_function.this[0]" my-function
module.lambda.aws_lambda_function.this[0]: Importing from ID "my-function"...
module.lambda.aws_lambda_function.this[0]: Import prepared!
  Prepared aws_lambda_function for import
module.lambda.aws_lambda_function.this[0]: Refreshing state... [id=my-function]

Error: Error in function call

  on .terraform/modules/lambda/iam.tf line 9, in locals:
   9:   role_name = local.create_role ? coalesce(var.role_name, var.function_name) : null
    |----------------
    | var.function_name is ""
    | var.role_name is null

Call to function "coalesce" failed: no non-null, non-empty-string arguments.
```

Cleared by switching to v1.28.0, which does not have the line with the error.

**Comment by antonbabenko** [comment]
You need to specify some value for either `function_name` or `role_name` inside of `module` block before calling `terraform import`, like this:

```terraform
module "lambda" {
  source = "terraform-aws-modules/lambda/aws"

  function_name = "my-function"
  handler       = "index.lambda_handler"
  runtime       = "python3.8"
}
```

There is not much we can do in this module, unfortunately!

**Comment by blbradley** [comment]
Function name and role name were already defined.

**Comment by antonbabenko** [comment]
I see the error you have when I don't have `function_name` argument in the module block and it works successfully when I specify it.

**Comment by bmurphey** [comment]
I confirmed I'm getting the same results, with Terraform 0.12.30 and module v1.35.0.  Switching to module v1.28.0 works.

My code:
```
module "main-lambda-trigger" {
  source = "terraform-aws-modules/lambda/aws"
  version = "1.35"

  function_name = "trigger"
  description   = "My awesome lambda function"
  handler       = "trigger.lambda_handler"
  runtime       = "python3.8"

  create_package = false

  tags = {
    Name = "trigger"
  }
}
```

Specifying `role_name` doesn't help, either.

**Comment by mike-carey** [comment]
I have created a PR that could potentially worked around this.  The other option I was able to use was to pin it to a lower version to do the imports and then raise the version back.

**Comment by antonbabenko** [comment]
This should be now fixed and available in [v1.43.0](https://github.com/terraform-aws-modules/terraform-aws-lambda/releases/v1.43.0) of this module (see #121), which by the way has the minimum Terraform version requirements of 0.12.26.
